File: php/pro/test/base/test_client_resolve.php
<?php
namespace ccxt\pro;
include_once (__DIR__.'/../../../../ccxt.php');


use React\Async;
use React\sleep;


function create_and_watch_orders($exchange, $symbol) {
    return Async\async(function () use ($exchange, $symbol) {
        yield sleep(10);
        $order1 = yield $exchange->create_order($symbol, 'limit', 'buy', 10, 0.5);
        echo 'createOrder:', $order1['id'], "\n";

        $order2 = yield $exchange->create_order($symbol, 'limit', 'buy', 11, 0.49);
        echo 'createOrder:', $order2['id'], "\n";

        $order3 = yield $exchange->create_order($symbol, 'limit', 'buy', 11, 0.48);
        echo 'createOrder:', $order3['id'], "\n";

        $orders = yield $exchange->watch_orders($symbol);
        echo 'orders:', count($orders), "\n";

        assert(count($orders) === 2, 'Expecting 2 orders to be returned');

        $exchange->close();
    })();
}

$test_create_and_watch_orders = function () {
    $apiKey = 'dcdb042f4f0603265bce20abab5ca557016b0d0103010850f80faf40c8a747c0';
    $secret = '0359f4ecba2509781c73847786c485d934780163f9f6a8bb89b55617b939b77a';

    $exchangeClass = '\\ccxt\\pro\\binanceusdm';
    $exchange = new $exchangeClass([
        'apiKey' => $apiKey,
        'secret' => $secret,
        'enableRateLimit' => true
    ]);

    $exchange->set_sandbox_mode(true);
    $exchange->verbose = true;
    $symbol = 'XRP/USDT:USDT';

    Async\await(create_and_watch_orders($exchange, $symbol));
};

\React\Async\coroutine($test_create_and_watch_orders);

File: python/ccxt/pro/test/base/test_client_resolve.py
import os
import sys

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))))
sys.path.append(root)

import asyncio
import ccxt.pro
import os

async def sleep(seconds):
    await asyncio.sleep(seconds)

async def main():
    # Configure the API and secret keys
    api_key = ''
    secret = ''

    # Initialize the exchange
    exchange = ccxt.pro.binanceusdm({
        'apiKey': api_key,
        'secret': secret
    })

    # Set to use the sandbox (testnet) environment
    exchange.set_sandbox_mode(True)
    symbol = 'XRP/USDT:USDT'

    # Watch orders (not directly translatable to Python; need to handle asynchronously)
    asyncio.create_task(exchange.watch_orders(symbol))

    # Simulate asynchronous order creation
    await sleep(10)
    order = await exchange.create_order(symbol, 'limit', 'buy', 10, 0.5)
    print('createOrder:', order['id'])

    await exchange.create_order(symbol, 'limit', 'buy', 11, 0.49)
    print('createOrder:', order['id'])

    await exchange.create_order(symbol, 'limit', 'buy', 11, 0.48)
    print('createOrder:', order['id'])

    # Watch orders for the symbol
    orders = await exchange.watch_orders('XRP/USDT')
    print('orders:', len(orders))

    # Use assertion to validate the number of orders
    assert len(orders) == 2, 'expecting 2 orders to be returned'

    # Close the connection
    await exchange.close()

# Run the main function
asyncio.run(main())
